.github/workflows/check-urls.yml

```diff
@@ -0,0 +1,161 @@
+name: Check URLs
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '17 5 * * 0' # 5:17 AM every Sunday
+  pull_request:
+    branches: [ main ]
+env:
+  ignore_file_patterns: |
+    docs
+    images
+    utils
+    Events
+
+jobs:
+  check-urls:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Install dependencies
+      run: |
+        python -m pip --no-cache-dir --disable-pip-version-check install --upgrade pip
+        python -m pip --no-cache-dir --disable-pip-version-check install linkchecker
+
+    - name: Reformat environment variables
+      id: setup_vars
+      run: |
+        tmp=$(echo "${{ env.ignore_file_patterns }}" | tr '\n' ' ')
+        echo "ignore_file_patterns=$tmp" >> $GITHUB_OUTPUT
+
+    - name: Checkout Repo for PR branch
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: actions/checkout@v4
+
+    - name: Checkout Repo for link check
+      if: ${{ github.event_name != 'pull_request' }}
+      uses: actions/checkout@v4
+      with:
+        ref: 'sched-link-checks'
+
+    - name: Sync main to link check branch
+      if: ${{ github.event_name != 'pull_request' }}
+      run: | 
+        git config user.name 'github-actions'
+        git config user.email 'github-actions@github.com'
+        git fetch origin main
+        git merge origin/main --no-edit -X thiers
+        git push origin sched-link-checks
+
+    - name: Get Changed Files (for PRs)
+      if: ${{ github.event_name == 'pull_request' }}
+      id: changed-files
+      uses: tj-actions/changed-files@v42
+      with:
+        separator: ' '
+
+    - name: Generate lists of files to check and ignore
+      id: file_list
+      run: |
+        if [ "${{ github.event_name }}" = "pull_request" ]; then
+          echo "files=${{ steps.changed-files.outputs.all_changed_files }}" >> $GITHUB_OUTPUT
+          echo "ignore_file_patterns=" >> $GITHUB_OUTPUT
+        else
+          echo "files=" >> $GITHUB_OUTPUT
+          echo "ignore_file_patterns=${{ steps.setup_vars.outputs.ignore_file_patterns }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Check URLs in selected files
+      run: |
+        for f in ${{ steps.file_list.outputs.files }}; do
+            if [ "${f##*.}" != "md" ]; then
+                continue
+            fi
+            for ef in ${{ steps.file_list.outputs.ignore_file_patterns }}; do
+                if [ "$ef" = "$f" ]; then
+                    continue 2 # ignore this file
+                fi
+                fd=$(echo $f | cut -d'/' -f1)
+                if [ "$ef" = "$fd" ]; then
+                    continue 2 # ignore this dir
+                fi
+            done
+            linkchecker -f utils/LinkChecker/.linkcheckerrc file://$(pwd)/$f >> linkchecker.out || true
+            cat linkchecker.out >> linkchecker-all.out
+        done
```

Comment:
Seems like it might be good to break this out into a Python script taking arguments so it can be developed and tested locally.  Also, the Python implementation for some of these operations is a bit cleaner than bash commands.